src/js/Scroll.js

```diff
@@ -12,27 +14,34 @@ class Scroll {
 
     this.$header = $('#header');
 
-    $(document).ready(this.onReady.bind(this));
+    // BINDS
+    this.onReady = this.onReady.bind(this);
+    this.onScroll = this.onScroll.bind(this);
+    this.setScrollOffset = this.setScrollOffset.bind(this);
+    this.checkAndScroll = this.checkAndScroll.bind(this);
```

Comment:
@varffs this way the bound function is created once and not every time an event triggers